platforms/quorum/configuration/roles/create/generate_crypto_tessera/tasks/main.yaml

```diff
@@ -0,0 +1,49 @@
+# This task checks if the tm crypto material is already stored in the vault or not
+- name: Check if tm key is already present in the vault
+  shell: |
+    vault read -field=tm.key secret/{{ component_ns }}/crypto/{{ component_name }}/transaction/
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  register: vault_tmkey_result
+  ignore_errors: yes
+
+# This task creates the build directory if it does not exist
+- name: Create build directory if it does not exist
+  file:
+    path: build/{{ component_name }}
+    state: directory
+    mode: '0755'
+    recurse: yes
+  when: vault_tmkey_result.failed == True 
+
+# This task checks if the tessera app jar exits
+- name: Check if tessera jar file exists
+  stat: 
+    path: build/tessera-app.jar
+  register: tessera_app_jar_result
+  when: vault_tmkey_result.failed == True
+
+# This task downloads the tessera jar file
+- name: Download tessera jar
+  shell: |
+    cd build
+    wget -q https://github.com/jpmorganchase/tessera/releases/download/tessera-0.8/tessera-app-0.8-app.jar
+    mv tessera-app-0.8-app.jar tessera-app.jar
+  when: vault_tmkey_result.failed == True and tessera_app_jar_result.stat.exists == False
+
+# This task generates keys for node tm
+- name: Generate node tm keys
+  shell: |
+    cd build/{{ component_name }}
+    printf '\n\n' | java -jar ../tessera-app.jar -keygen -filename tm
+  when: vault_tmkey_result.failed == True
+
+# This task copies the crypto material to the vault
+- name: Copy the crypto material to Vault
+  shell: |
+    vault kv put secret/{{ component_ns }}/crypto/{{ component_name }}/transaction tm.pub="$(cat build/{{ component_name }}/tm.pub)" tm.key="$(cat build/{{ component_name }}/tm.key)"
```

Comment:
The path will change {{ component_name }} shall be replaced to {{ peer.name }} with a loop over peers. I am currently doing this change for RAFT